cmd/infracost/run.go

```diff
@@ -258,20 +258,23 @@ func loadRunFlags(cfg *config.Config, cmd *cobra.Command) error {
 	cfg.ShowSkipped, _ = cmd.Flags().GetBool("show-skipped")
 	cfg.SyncUsageFile, _ = cmd.Flags().GetBool("sync-usage-file")
 
+	includeAllFields := "all"
 	validFields := []string{"price", "monthlyQuantity", "unit", "hourlyCost", "monthlyCost"}
 	validFieldsFormats := []string{"table", "html"}
 
 	if cmd.Flags().Changed("fields") {
-		if c, _ := cmd.Flags().GetStringSlice("fields"); len(c) == 0 {
+		fields, _ := cmd.Flags().GetStringSlice("fields")
+		if len(fields) == 0 {
 			ui.PrintWarningf("fields is empty, using defaults: %s", cmd.Flag("fields").DefValue)
 		} else if cfg.Fields != nil && !contains(validFieldsFormats, cfg.Format) {
```

Comment:
Did you happen to look into how this else if works (and if it makes sense with the 'all' option)?  I'm not 100% familiar with this so I can check it out if you haven't.  I'm guessing that cfg.Fields can get populated from a config file so we might need to support 'all' there too.

Reply:
We don't support fields in the config file yet. The config file fields are defined [here](https://github.com/infracost/infracost/blob/master/internal/config/config_file.go#L16-L19) and [here](https://github.com/infracost/infracost/blob/master/internal/config/config.go#L15-L24).

Side note: not sure why we have [YAML defs for other config values](https://github.com/infracost/infracost/blob/master/internal/config/config.go#L26-L45) - we should remove these if not used so it's not confusing.